src/interpreter/fvm2.rs

```diff
@@ -72,20 +74,27 @@ impl<DB: Blockstore> ForestExternsV2<DB> {
             );
         }
 
-        let prev_root = (self.lookback)(height)?;
-        let lb_state = StateTree::new_from_root(&self.db, &prev_root)?;
+        let prev_root = {
+            let (_, st) = self.chain_store.get_lookback_tipset_for_round(
+                Arc::clone(&self.chain_config),
+                Arc::clone(&self.heaviest_tipset),
+                height,
+            )?;
+            st
+        };
+        let lb_state = StateTree::new_from_root(&self.chain_store.db, &prev_root)?;
```

Comment:
should we take this out into a `get_lb_state` method?

Reply:
Yes!